### ISSUE TITLE
Regression on 2.0.47.0 for DirectoryInfo

### ISSUE THREAD

**Issue by azchohfi** [peteraritchie/LongPath#82]
This commit changed a behavior on DirectoryInfo.Exist:
https://github.com/peteraritchie/LongPath/commit/756979131828c796ca77688fcf5390d8a53ce43f

On 2.0.45.0, if you try to:
```
var di = new DirectoryInfo(longPathFilename);
Assert.IsFalse(di.Exists);
```
On 2.0.47.0, it throws an exception.
This is only if the file exists. Unfortunately, this is not in the DirectoryInfo test cases.

**Comment by azchohfi** [comment]
I'm having a hard time on a possible solution for this... I can't break the NormalizedPathWithSearchPatternIncludesWildcardForFolders test...
What should the right approach be @peteraritchie ?

**Comment by peteraritchie** [comment]
This is fixed in 2.0.51
